src/Controllers/GagePageController.ts

```diff
@@ -673,17 +694,156 @@ module StreamStats.Controllers {
                         dailyValues = 0
                     };
                     this.dailyFlow = dailyValues
-                    this.formatData();
+                    this.getRatingCurve();
                 }); 
             }
 
+        public getRatingCurve() {
+            const url = 'https://waterdata.usgs.gov/nwisweb/get_ratings?site_no=' + this.gage.code + '&file_type=exsa'
+            // console.log('getDischargeInfo', url)
+            const request: WiM.Services.Helpers.RequestInfo = new WiM.Services.Helpers.RequestInfo(url, true, WiM.Services.Helpers.methodType.GET, 'json');
+            
+            this.dischargeObj = [];
+            // console.log('discharge data', this.dischargeObj)
+
+            this.Execute(request).then(
+                (response: any) => {
+                    // console.log('response?', response)
+                    const data = response.data.split('\n').filter(r => { return (!r.startsWith("#") && r != "") });
+                    //console.log('data', data)
+                    data.shift().split('\t');
+                    //console.log('data with shift', data)
+                    data.shift();
+                    // console.log('discharge data', dischargeObj)
+                    data.forEach(row => {
+                        let dataRow = row.split('\t')
+                        const object = {
+                            x: parseFloat(dataRow[2]),
+                            y: parseFloat(dataRow[0])
+                        };
+                        // console.log('ratingcurve data', dataRow)
+                        this.dischargeObj.push(object) 
+                   });
+                   // console.log('this.discharge obj 1st one', this.dischargeObj)
+                    // console.log('dischargeObj', dischargeValue)
+                }, (error) => {
+                    // console.log(error)
+                    // this.ratingCurve = this.dischargeObj;
+                }).finally(() => {
+                    this.getUSGSMeasured()
+                });
+        }       
+
+        public getUSGSMeasured() {
+            const url = 'https://waterdata.usgs.gov/nwis/measurements?site_no=' + this.gage.code + '&agency_cd=USGS&format=rdb_expanded'
+            // console.log('usgsMeasuredURL', url)
+            const request: WiM.Services.Helpers.RequestInfo = new WiM.Services.Helpers.RequestInfo(url, true, WiM.Services.Helpers.methodType.GET, 'json');
+            
+            this.measuredObj = [];
+            // console.log('is measured obj have data', this.measuredObj)
+            this.Execute(request).then(
+                (response: any) => {
+                    const data = response.data
+                    //console.log('data error message', data)
+                    var errorMessage = '<title>USGS NwisWeb error message</title>'
+                    this.error = data.includes(errorMessage)
```

Comment:
If you change the `showInLegend` line below, then you can make this a local variable and then we could remove (& ignore my comment on) the global `error` variable above.

Reply:
Again, just not sure if I should remove the global error variable since it's used in other parts of the code that I didn't write. Let me know otherwise!